dot/core/service_integration_test.go

```diff
@@ -37,6 +38,98 @@ import (
 
 //go:generate mockgen -destination=mock_telemetry_test.go -package $GOPACKAGE github.com/ChainSafe/gossamer/dot/telemetry Client
 
+func balanceKey(t *testing.T, pub []byte) []byte {
+	h0, err := common.Twox128Hash([]byte("System"))
+	require.NoError(t, err)
+	h1, err := common.Twox128Hash([]byte("Account"))
+	require.NoError(t, err)
+	h2, err := common.Blake2b128(pub)
+	require.NoError(t, err)
+	return append(append(append(h0, h1...), h2...), pub...)
+}
+
+func newTestDigest(t *testing.T) scale.VaryingDataTypeSlice {
+	vdts := types.NewDigest()
+	err := vdts.Add(
+		types.PreRuntimeDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              common.MustHexToBytes("0x0201000000ef55a50f00000000"),
+		},
+		types.ConsensusDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              common.MustHexToBytes("0x0118ca239392960473fe1bc65f94ee27d890a49c1b200c006ff5dcc525330ecc16770100000000000000b46f01874ce7abbb5220e8fd89bede0adad14c73039d91e28e881823433e723f0100000000000000d684d9176d6eb69887540c9a89fa6097adea82fc4b0ff26d1062b488f352e179010000000000000068195a71bdde49117a616424bdc60a1733e96acb1da5aeab5d268cf2a572e94101000000000000001a0575ef4ae24bdfd31f4cb5bd61239ae67c12d4e64ae51ac756044aa6ad8200010000000000000018168f2aad0081a25728961ee00627cfe35e39833c805016632bf7c14da5800901000000000000000000000000000000000000000000000000000000000000000000000000000000"), //nolint:lll
```

Comment:
where is this from? can you generate it instead of hardcoding it?

Reply:
hmmm I got this digest from the types/digest_test file and it isn't marked there where they are from. Do you have a suggestion of how I could generate this?

Reply:
I think you can just generate any random consensus digest and put it in here, it doesn't even look like this is a real digest lol. if the data doesn't matter you could also just put some random bytes

Reply:
ohhh I think they are hardcoded BabePreDigests. They do need to be valid, its updated to generate a babePreDigest now rather than just have a hex string

Reply:
let me know if this should be fixed further!